Source/DSPTool/DSPTool.cpp

```diff
@@ -246,42 +271,30 @@ static bool PerformAssembly(const std::string& input_name, const std::string& ou
   }
 
   std::string source;
-  if (File::ReadFileToString(input_name.c_str(), source))
+  if (File::ReadFileToString(input_name, source))
   {
     if (multiple)
     {
+      source.append("\n");
+
       // When specifying a list of files we must compile a header
       // (we can't assemble multiple files to one binary)
       // since we checked it before, we assume output_header_name isn't empty
-      int lines;
-      std::vector<u16>* codes;
-      std::vector<std::string> files;
-      std::string header, currentSource;
-      size_t lastPos = 0, pos = 0;
-
-      source.append("\n");
-
-      while ((pos = source.find('\n', lastPos)) != std::string::npos)
-      {
-        std::string temp = source.substr(lastPos, pos - lastPos);
-        if (!temp.empty())
-          files.push_back(temp);
-        lastPos = pos + 1;
-      }
-
-      lines = (int)files.size();
+      std::string currentSource;
+      const std::vector<std::string> files = GetAssemblerFiles(source);
 
+      std::size_t lines = files.size();
       if (lines == 0)
       {
         printf("ERROR: Must specify at least one file\n");
         return false;
       }
 
-      codes = new std::vector<u16>[lines];
+      std::vector<std::vector<u16>> codes(lines);
 
-      for (int i = 0; i < lines; i++)
+      for (std::size_t i = 0; i < lines; i++)
```

Comment:
Would this be a candidate for a range-based for loop?

Reply:
Nope, because `files[i]` and `codes[i]` are both used in the loop.

Reply:
Missed that one :/
My brain kinda assumed `files[i]` for the `DSP::Assemble` call for whatever reason...